Fix #9417: No difficulty level to select for questions while creating question suggestion

sagangwee opened the pull request:
## Overview
<!--
READ ME FIRST:
Please answer *both* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

1. This PR fixes #9417 .
2. This PR fixes the missing difficulty selections in the question suggestion modal. See screenshot below:

<img width="1671" alt="Screen Shot 2020-06-04 at 10 08 46 PM" src="https://user-images.githubusercontent.com/9004726/83841231-4f6c2400-a6b5-11ea-9be4-c49471930e8e.png">

Creating question in topic editor:
<img width="1678" alt="Screen Shot 2020-06-04 at 10 48 14 PM" src="https://user-images.githubusercontent.com/9004726/83841335-82aeb300-a6b5-11ea-99f1-b177de2fd8ec.png">

## Essential Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The linter/Karma presubmit checks have passed locally on your machine.
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
  - This lets reviewers restart your CircleCI tests for you.
- [x] The PR is made from a branch that's **not** called "develop".

## PR Pointers

- Oppiabot will notify you when you don't add a PR_CHANGELOG label. If you are unable to do so, please @-mention a code owner (who will be in the Reviewers list), or ask on [Gitter](https://gitter.im/oppia/oppia-chat).
- For what code owners will expect, see the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia%27s-code-owners-and-checks-to-be-carried-out-by-developers).
- Make sure your PR follows conventions in the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide), otherwise this will lead to review delays
- Never force push. If you do, your PR will be closed.

sagangwee committed:
skill difficulty selector ui fixes

sagangwee commented:
@DubeySandeep @aks681 PTAL. Thanks!

DubeySandeep commented:
@aks681 Can you please review this quickly as it's needed for the hotfix? Thanks!

aks681 commented on core/templates/components/question-difficulty-selector/question-difficulty-selector.directive.html:
```diff
@@ -1,14 +1,10 @@
   <md-radio-button ng-repeat="rubric in getSkillIdToRubricsObject()[skillWithDifficulty.getId()]" value="<[availableDifficultyValues[$index]]>">
     <strong> <[rubric.getDifficulty()]> </strong>
-    <md-list>
-      <md-list-item ng-repeat="explanation in rubric.getExplanations()">
```
Why is this removed? We have multiple explanations per rubric now and this directive is used elsewhere as well.

sagangwee replied:
The changes in #9148 caused the issues outlined [here](https://github.com/oppia/oppia/issues/9417#issuecomment-637717102), and the list elements were causing the formatting issues. I'd imagine since the topics and skills dashboard uses the same `select-skill-and-difficulty-modal` which uses the new `question-difficulty-selector` directive, the formatting issues are reflected in the topics and skills dashboard as well. Since this change is urgently needed for a hotfix, I opted for the simpler reversion here.

Since it seems the other current usage of the `question-difficulty-selector` directive is in the `questions-list` directive, do you know where that is used?

aks681 replied:
The formatting is like that because there are multiple elements, which is required.
I think it is used when creating a question and when changing the difficulty of a question in questions-list, so this change here shouldn't be done as multiple explanations per rubric is something we support and this would break the editors.
You can try fixing the formatting here, or if this is urgent, in a follow up PR and just add the 'include' below in this PR.

sagangwee commented:
@aks681 Responded to your comment. PTAL!

sagangwee committed:
revert explanation list

aks681 reviewed (approved):
Lgtm!

sagangwee commented:
@aks681 We can decide on styling later. I've just fixed the missing difficulty selection in the question suggestion for now and added back the list elements:

<img width="1680" alt="Screen Shot 2020-06-05 at 11 58 31 AM" src="https://user-images.githubusercontent.com/9004726/83922825-e5479380-a735-11ea-8402-d34b25c37819.png">
